src/pygeoprocessing/routing/routing.pyx

```diff
@@ -2960,3 +3006,1298 @@ def _is_raster_path_band_formatted(raster_path_band):
+
+    cdef int edge_side, edge_dir, cell_to_test, out_dir_increase=-1
+    cdef int left, right, n_left_turns
+
+    for index, (stream_fid, x_l, y_l) in enumerate(visit_order_stack):
+        if ctime(NULL) - last_log_time > _LOGGING_PERIOD:
+            LOGGER.info(
+                f'(calculate_subwatershed_boundary): watershed building '
+                f'{(index/len(visit_order_stack))*100:.1f}% complete')
+            last_log_time = ctime(NULL)
+        discovery = <long>discovery_managed_raster.get(x_l, y_l)
+        if discovery == -1:
+            continue
+        boundary_list = [(x_l, y_l)]
+        finish = <long>finish_managed_raster.get(x_l, y_l)
+
+        watershed_boundary = ogr.Geometry(ogr.wkbLinearRing)
+        outflow_dir = <int>d8_flow_dir_managed_raster.get(x_l, y_l)
+
+        # this is the center point of the pixel that will be offset to
+        # make the edge
+        x_f = x_l+0.5
+        y_f = y_l+0.5
+
+        x_f += D8_XOFFSET[outflow_dir]*0.5
+        y_f += D8_YOFFSET[outflow_dir]*0.5
+        if outflow_dir % 2 == 0:
+            # need to back up the point a bit
+            x_f -= D8_YOFFSET[outflow_dir]*0.5
+            y_f += D8_XOFFSET[outflow_dir]*0.5
+
+        x_p, y_p = gdal.ApplyGeoTransform(geotransform, x_f, y_f)
+        watershed_boundary.AddPoint(x_p, y_p)
+        x_first, y_first = x_p, y_p
+
+        # determine the first edge
+        if outflow_dir % 2 == 0:
+            # outflow through a straight side, so trivial edge detection
+            edge_side = outflow_dir
+            edge_dir = (2+edge_side) % 8
+        else:
+            # diagonal outflow requires testing neighboring cells to
+            # determine first edge
+            cell_to_test = (outflow_dir+1) % 8
+            edge_side = cell_to_test
+            edge_dir = (cell_to_test+2) % 8
+            if _in_watershed(
+                    x_l, y_l, cell_to_test, discovery, finish,
+                    n_cols, n_rows,
+                    discovery_managed_raster, discovery_nodata):
+                edge_side = (edge_side-2) % 8
+                edge_dir = (edge_dir-2) % 8
+                x_l += D8_XOFFSET[edge_dir]
+                y_l += D8_YOFFSET[edge_dir]
+                # note the pixel moved
+                boundary_list.append((x_l, y_l))
+
+        # use to keep track of how many turns around a pixel, used if
+        # a degerate
+        n_left_turns = 0
+        while True:
+            # step the edge then determine the projected coordinates
+            x_f += D8_XOFFSET[edge_dir]
+            y_f += D8_YOFFSET[edge_dir]
+            # equivalent to gdal.ApplyGeoTransform(geotransform, x_f, y_f):
+            x_p = g0 + g1*x_f + g2*y_f
+            y_p = g3 + g4*x_f + g5*y_f
+            watershed_boundary.AddPoint(x_p, y_p)
+
+            if n_left_turns == 4:
+                LOGGER.info(
+                    '4 left turns around the same pixel so terminating')

```

Comment:
Would it be useful to log some other information about this feature, such as the FID of the watershed?

Reply:
That line is no longer in there because i no longer explicitly check for 4 turns as in 28fbda7 above